src/unity/Assets/Crest/Scripts/Collision/CollProviderDispTexs.cs

```diff
@@ -13,25 +13,31 @@ public class CollProviderDispTexs : ICollProvider
     {
         int _areaLod = -1;
 
-        public bool SampleDisplacement(ref Vector3 worldPos, ref Vector3 displacement)
+        public bool SampleDisplacement(ref Vector3 in__worldPos, out Vector3 displacement)
         {
-            int lod = LodDataAnimatedWaves.SuggestDataLOD(new Rect(worldPos.x, worldPos.z, 0f, 0f), 0f);
-            if (lod == -1) return false;
-            return OceanRenderer.Instance._lodDataAnimWaves[lod].SampleDisplacement(ref worldPos, ref displacement);
+            int lod = LodDataAnimatedWaves.SuggestDataLOD(new Rect(in__worldPos.x, in__worldPos.z, 0f, 0f), 0f);
+            if (lod == -1) {
+                displacement = Vector2.zero;
```

Comment:
Should this be Vector3.zero instead of Vector2.zero

Reply:
(theres a second Vector2 typo in one other place as well)

Reply:
You are correct, it's annoying that unity autocasts like that